src/core_codemods/defectdojo/api.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import cast
+
+from typing_extensions import override
+
+from codemodder.codemods.api import Metadata, Reference, ToolMetadata
+from codemodder.codemods.base_detector import BaseDetector
+from codemodder.context import CodemodExecutionContext
+from codemodder.result import ResultSet
+from core_codemods.api import CoreCodemod, SASTCodemod
+
+from .results import DefectDojoResultSet
+
+
+class DefectDojoDetector(BaseDetector):
+    def apply(
+        self,
+        codemod_id: str,
+        context: CodemodExecutionContext,
+        files_to_analyze: list[Path],
+    ) -> ResultSet:
+        result_set = DefectDojoResultSet()
+        for filename in context.tool_result_files_map.get("defectdojo", []):
```

Comment:
I think you may also want a cache version here just like sonar has in `process_sonar_findings`